### PR TITLE
Comp/san antonio

### PR DESCRIPTION
**Pull request by KianChilds**
## Justification
Implementation of the changes requested at comp 

## Testing Done
Ran at comp

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -237,7 +242,14 @@ private void configureBindings() {
     OI.getButton(OI.Operator.retractClimber).toggleOnTrue(climberCommandFactory.climb());
 
     new Trigger(() -> OI.Operator.controller.getPOV() == 0).whileTrue(intakeCommand());
+    new Trigger(() -> OI.Operator.controller.getPOV() == 270)
+        .onTrue(new InstantCommand(() -> robotStateManager.setLongRange()));
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/RobotContainer.java]: Is this the most recent code?  I seem to remember that longrange was unmapped from the commands. @KianChilds

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -237,7 +240,13 @@ private void configureBindings() {
     OI.getButton(OI.Operator.retractClimber).toggleOnTrue(climberCommandFactory.climb());
 
     new Trigger(() -> OI.Operator.controller.getPOV() == 0).whileTrue(intakeCommand());
+    // new Trigger(() -> OI.Operator.controller.getPOV() == 270)
+    //     .onTrue(new InstantCommand(() -> robotStateManager.setLongRange()));
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/RobotContainer.java:244]: Let's actually remove this code, and the setShortRange

**Review by azvanderpas — CHANGES_REQUESTED** [review]
Address comments from #214

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -252,24 +261,42 @@ private Command intakeCommand() {
...
   private Command shooterOuttake() {
-
     return Commands.parallel(
             triggerCommandFactory.getEjectCommand(), intakeCommandFactory.reverseIntakeCommand())
         .asProxy();
   }
 
   private void configDriverFeedBack() {
     new Trigger(trapElvCommandFactory.getSourceBreak())
-        .and(OI.getTrigger(OI.Driver.intake))
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/RobotContainer.java:263]: We are checking the OI.Driver.intake trigger twice, once to do the intake and another to trigger rumble.

We should see if we can include this in the intake command, or in the command group that is attached to the Intake trigger.

Also, we have a signaling subsystem that has rumble.  Should we use that instead?

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -279,42 +306,71 @@ private Command speakerSource() {
...
   private Command shootSpeaker() {
     return Commands.parallel(
-        Commands.either(
-            triggerCommandFactory.getShootCommand(),
-            triggerCommandFactory
-                .getShootCommand()
-                .onlyIf(() -> shooterCommandFactory.isShooterReady())
-                .asProxy(),
-            OI.getButton(OI.Operator.simple)),
+        Commands.waitSeconds(CommandConstants.WAIT_FOR_TRAPELV)
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/RobotContainer.java:322]: What are we waiting for here?  Is this to make sure that isShooterReady is stable?  If so, we should use a debounce instead

File: src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java [hunk 5]
```diff
@@ -53,8 +53,8 @@ private void configMotor(CANSparkMax motor, boolean invert) {
     pidController.setD(ClimberConstants.CURRENT_PID[2]);
     double armPosition;
 
-    armPosition = motor.getAbsoluteEncoder().getPosition();
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java:56]: So we disabled the absolute encoder during competition? 
Was there a reason why? Are we not expecting to use the absolute encoder in district champs?

File: src/main/java/frc/robot/RobotContainer.java (hunk 4 above)
Comment (KianChilds) [reply src/main/java/frc/robot/RobotContainer.java:322]: we are waiting for the trapelv to move out of the way

File: src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java (hunk 5 above)
Comment (KianChilds) [reply src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java:56]: it was occasionally placing the arms at large negative or positive values. Can be reimplemented when needed

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (KianChilds) [reply src/main/java/frc/robot/RobotContainer.java:263]: to both points: possible, out of scope of this PR

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (KianChilds) [reply src/main/java/frc/robot/RobotContainer.java]: resolved

File: src/main/java/frc/robot/utilities/AutoManager.java [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/utilities/AutoManager.java:1]: DElete this file

**Review by azvanderpas — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 7]
```diff
@@ -301,39 +349,36 @@ private static double fineTuneTurretRotation(
...
   public void setTurretPos(double setpoint) {
     if (!Constants.enabledSubsystems.turretRotationEnabled) return;
+    double limitedSetpoint =
+        MathUtil.clamp(
+            setpoint,
+            Math.toRadians(Constants.TurretConstants.TURRET_MIN_ANGLE_DEGREES),
+            Math.toRadians(Constants.TurretConstants.TURRET_MAX_ANGLE_DEGREES));
+    turretGoalPositionEntry.log(Units.radiansToDegrees(setpoint));
+    this.setPositionErrorSupplier(() -> turretPosition - limitedSetpoint);
+  }
```
[the remark this answers is not in the document]
Comment (KianChilds) [reply src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java]: besides controlling the PID through setpoint. Here it is controlled by the error function of the PID. the PIDs goal is always zero making whatever number is returned by the error function the negative of the error processed by the PID.

**Comment by caseyjbrotherton** [comment]
This shouldn't be needed anymore now that newer branches have been merged

**Pull request closed without merging**